File: Backend/FitTrackWithMLP/FitTrackWithMLP.Shared/DTOs/DailyPlan/PlannedMealDto.cs
namespace FitTrackWithMLP.Shared.DTOs.DailyPlan
{
    public class PlannedMealDto
    {
        public int PlannedMealId { get; set; }
        public string Category { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public List<PlannedMealIngredientDto> Ingredients { get; set; } = new();
        // Total calories are computed on the fly via AutoMapper
        public int TotalCalories { get; set; }
    }
}

File: Frontend/src/dtos/DailyPlan/Get/PlannedMealDto.ts
import type { PlannedMealIngredientDto } from "./PlannedMealIngredientDto";

export interface PlannedMealDto {
    plannedMealId: number;
    category: string;
    title: string;
    ingredients: PlannedMealIngredientDto[];
    totalCalories: number;
}
